ledger/eval_test.go

```diff
@@ -318,3 +323,238 @@ func TestPrepareAppEvaluators(t *testing.T) {
+	deadlockDisable := deadlock.Opts.Disable
+	deadlock.Opts.Disable = true
+	defer func() { deadlock.Opts.Disable = deadlockDisable }()
+	start := time.Now()
+	genesisInitState, addrs, keys := genesis(100000)
+	dbName := fmt.Sprintf("%s.%d", b.Name(), crypto.RandUint64())
+	proto := config.Consensus[genesisInitState.Block.CurrentProtocol]
+	proto.MaxTxnBytesPerBlock = 1000000000 // very big, no limit
+	config.Consensus[protocol.ConsensusVersion(dbName)] = proto
+	genesisInitState.Block.CurrentProtocol = protocol.ConsensusVersion(dbName)
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	l, err := OpenLedger(logging.Base(), dbName, inMem, genesisInitState, cfg)
+	require.NoError(b, err)
+	defer testLedgerCleanup(l, dbName, inMem)
+
+	dbName2 := dbName + "_2"
+	l2, err := OpenLedger(logging.Base(), dbName2, inMem, genesisInitState, cfg)
+	require.NoError(b, err)
+	defer testLedgerCleanup(l2, dbName2, inMem)
+
+	setupDone := time.Now()
+	setupTime := setupDone.Sub(start)
+	b.Logf("BenchmarkBlockEvaluator setup time %s", setupTime.String())
+
+	bepprof := os.Getenv("BLOCK_EVAL_PPROF")
+	if len(bepprof) > 0 {
+		profpath := dbName + "_cpuprof"
+		profout, err := os.Create(profpath)
+		if err != nil {
+			b.Fatal(err)
+			return
+		}
+		pprof.StartCPUProfile(profout)
+		defer func() {
+			pprof.StopCPUProfile()
+			profout.Close()
+		}()
+	}
+
+	b.ResetTimer()
+
+	// test speed of block building
+	newBlock := bookkeeping.MakeBlock(genesisInitState.Block.BlockHeader)
+	bev, err := l.StartEvaluator(newBlock.BlockHeader, 0)
+	require.NoError(b, err)
+
+	genHash := genesisInitState.Block.BlockHeader.GenesisHash
+
+	for i := 0; i < b.N; i++ {
+		sender := i % len(addrs)
+		receiver := (i + 1) % len(addrs)
+		txn := transactions.Transaction{
+			Type: protocol.PaymentTx,
+			Header: transactions.Header{
+				Sender:      addrs[sender],
+				Fee:         minFee,
+				FirstValid:  newBlock.Round(),
+				LastValid:   newBlock.Round(),
+				GenesisHash: genHash,
+			},
+			PaymentTxnFields: transactions.PaymentTxnFields{
+				Receiver: addrs[receiver],
+				Amount:   basics.MicroAlgos{Raw: 100},
+			},
+		}
+		st := txn.Sign(keys[sender])
+		err = bev.Transaction(st, transactions.ApplyData{})
+		require.NoError(b, err)
+	}
+
+	validatedBlock, err := bev.GenerateBlock()
+	require.NoError(b, err)
+
+	blockBuildDone := time.Now()
+	blockBuildTime := blockBuildDone.Sub(setupDone)
+	b.ReportMetric(float64(blockBuildTime)/float64(b.N), "ns/block_build_tx")
+
+	l.AddValidatedBlock(*validatedBlock, agreement.Certificate{})
+
+	avbDone := time.Now()
+	avbTime := avbDone.Sub(blockBuildDone)
+	b.ReportMetric(float64(avbTime)/float64(b.N), "ns/AddValidatedBlock_tx")
+
+	// test speed of block validation
+	// This should be the same as the eval line in ledger.go AddBlock()
+	// This is pulled out to isolate eval() time from db ops of AddValidatedBlock()
+	_, err = eval(context.Background(), l2, validatedBlock.blk, false, nil, nil)
```

Comment:
you've passed false to "validate" here.. I'm pretty sure that you're not going to make any validation here.

Reply:
turned out to be a unused copy of a prior version of the benchmark function. deleted.